packages/workers-shared/router-worker/src/worker.ts

```diff
@@ -128,6 +128,26 @@ export default {
 						}
 					}
 
+					if (url.pathname === "/_image") {
+						const hrefParam = url.searchParams.get("href");
+						if (hrefParam && hrefParam.match(/^\/\/.+/)) {
+							try {
+								const hrefUrl = new URL(
+									hrefParam.startsWith("//") ? "https:" + hrefParam : hrefParam
+								);
```

Comment:
This code is only executed if `hrefParam.match(/^\/\/.+/)` (l133)

```suggestion
								const hrefUrl = new URL("https:" + hrefParam);
```

Also  `hrefParam.length > 2 && hrefParam.startsWith("//")` could fe faster than `hrefParam.match(/^\/\/.+/)`